File: src/Baekjoon/Sorting/p1920.java
package Baekjoon.Sorting;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class p1920 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder bw = new StringBuilder();

        // 1. 입력받기
        int N = Integer.parseInt(br.readLine());
        int[] stNum = new int[N];   // standard numbers
        String[] buffer = br.readLine().split(" ");

        for (int i = 0; i < N; i++) {
            stNum[i] = Integer.parseInt(buffer[i]);
        }

        int M = Integer.parseInt(br.readLine());
        int[] cpNum = new int[M];   // compared numbers
        buffer = br.readLine().split(" ");

        for (int i = 0; i < M; i++) {
            cpNum[i] = Integer.parseInt(buffer[i]);
        }

        // 2. stNum 배열 오름차순 정렬 - 이분탐색을 쓰기 위해서
        Arrays.sort(stNum);

        // 3. 이분탐색으로 stNum에 cpNum이 있는지 찾기
        for (int i = 0; i < M; i++) {
            if (Arrays.binarySearch(stNum, cpNum[i]) >= 0) {
                bw.append("1\n");
            }
            else {
                bw.append("0\n");
            }
        }

        // 4. 답 출력
        System.out.print(bw);
    }
}

File: src/Baekjoon/p1920.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int N, M;
int A[100000];
int B[100000];

int main(){
    // 1. 입력받기
    cin >> N;

    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    cin >> M;

    for (int i = 0; i < M; i++) {
        cin >> B[i];
    }

    // 2. 배열 A 오름차순 정렬
    sort(A, A+N);
    
    // 3. B[i]가 A에 안에 있는지 이분탐색
    for (int i = 0; i < M; i++) {
        if (binary_search(A, A+N, B[i])) {
            cout << 1 << '\n';
        }
        else {
            cout << 0 << '\n';
        }
    }
}